<launch>
  <arg name="pcd_name" default="brown-cup-from-bag-other-rot" />
  <arg name="config_file" default="$(arg pcd_name)_tf.yaml" />
  <arg name="gui" default="true"/>

  <node name="static_tf_publisher_map_to_pcd"
        pkg="tf" type="static_transform_publisher"
        args="0 0 0 0 0 0 map $(arg pcd_name) 10"/>

  <node name="pointcloud_database_server" pkg="nodelet" type="nodelet"
        args="standalone jsk_pcl/PointcloudDatabaseServer" output="screen"
        clear_params="true">
    <rosparam subst_value="true">
      models: ["$(find jsk_2023_09_cook_from_recipe)/datas/pcds/$(arg pcd_name).pcd",
      ]
      duration: 0.05
      use_array: false
    </rosparam>
  </node>

  <!-- <node name="config_tf_broadcaster" pkg="known_object_ros" -->
  <!--       type="config_tf_publisher.py" output="screen" -->
  <!--       clear_params="true"> -->
  <!--   <rosparam subst_value="true"> -->
  <!--     config_file: $(arg config_file) -->
  <!--     parent_frame: map -->
  <!--     child_frame: target_object -->
  <!--   </rosparam> -->
  <!-- </node> -->

  <!-- <node name="itf" pkg="rviz_interactive_tf" type="interactive_tf" ns="itf1"> -->
  <!--   <param name="parent_frame" value="map"/> -->
  <!--   <param name="frame" value="tf1"/> -->
  <!-- </node> -->

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_2023_09_cook_from_recipe)/rviz/check_reference_pcd.rviz"/>
  </group>
</launch>
